src/main/java/net/dv8tion/jda/api/utils/MarkdownSanitizer.java

```diff
@@ -188,6 +188,123 @@ public static String escape(@Nonnull String sequence, int ignored)
+                    if(current == '>')
+                    {
+                        if(i+1 < sequence.length())
+                        {
+                            if(sequence.charAt(i+1) == ' ')
+                            {
+                                builder.append("\\>");
+                            }
+                            else
+                            {
+                                if(i + 3 < sequence.length())
+                                {
+                                    if(sequence.startsWith(">>> ", i))
+                                    {
+                                        builder.append("\\>\\>\\> ");
+                                        i += 3;
+                                        newline = false;
+                                        continue;
+                                    }
+                                }
+                                builder.append(current);
+                            }
+                            newline = false;
+                            continue;
+                        }
+                        else
+                            escaped = true;
+                    }
+                    else if(current == ' ')
+                    {
+                        builder.append(current);
+                        continue;
+                    }
+                    newline=false;
+                }
+
+                if(!escaped)
+                {
+                    switch (current)
+                    {
+                    case '*':
+                        builder.append("\\*");
+                        break;
+                    case '_':
+                        builder.append("\\_");
+                        break;
+                    case '`':
+                        builder.append("\\`");
+                        break;
+                    case '|':
+                        if(i+1 < sequence.length()){
+                            if(sequence.charAt(i+1) == '|'){
+                                builder.append("\\|\\|");
+                                i++;
+                                continue;
+                            }
+                        }
+                        builder.append("|");
+                        break;
+                    case '~':
+                        builder.append("\\~");
+                        break;
+                    case '\\':
+                        builder.append(current);
+                        escaped = true;
+                        break;
+                    case '\n':
+                        builder.append(current);
+                        newline = true;
+                        break;
+                    default:
+                        builder.append(current);
+                    }
+                }
+                else
+                {
+                    builder.append(current);
+                    escaped = false;
+                }
+
+            }
```

Comment:
I'm proposing a cleanup:

```suggestion
            for (int i = 0; i < sequence.length(); i++)
            {
                char current = sequence.charAt(i);
                if (newline)
                {
                    newline = Character.isWhitespace(current); // might still be a quote if prefixed by whitespace
                    if (current == '>')
                    {
                        // Check for quote if line starts with angle bracket
                        if (i + 1 < sequence.length() && Character.isWhitespace(sequence.charAt(i+1)))
                        {
                            builder.append("\\>"); // simple quote
                        }
                        else if (i + 3 < sequence.length() && sequence.startsWith(">>>", i) && Character.isWhitespace(sequence.charAt(i+3)))
                        {
                            builder.append("\\>\\>\\>").append(sequence.charAt(i+3)); // block quote
                            i += 3; // since we include 3 angle brackets AND whitespace
                        }
                        else
                        {
                            builder.append(current); // just a normal angle bracket
                        }
                        continue;
                    }
                }

                if (escaped)
                {
                    builder.append(current);
                    escaped = false;
                    continue;
                }
                // Handle average case
                switch (current)
                {
                case '*': // simple markdown escapes for single characters
                case '_':
                case '`':
                    builder.append('\\').append(current);
                    break;
                case '|': // cases that require at least 2 characters in sequence
                case '~':
                    if (i + 1 < sequence.length() && sequence.charAt(i+1) == current)
                    {
                        builder.append('\\').append(current)
                               .append('\\').append(current);
                        i++;
                    }
                    else
                        builder.append(current);
                    break;
                case '\\': // escape character
                    builder.append(current);
                    escaped = true;
                    break;
                case '\n': // linefeed is a special case for quotes
                    builder.append(current);
                    newline = true;
                    break;
                default:
                    builder.append(current);
                }
            }
```

The logic should be identical, I just rewrote it a bit and added some comments. Also, I replaced your check for `== ' '` with an `isWhitespace` which covers all possible space characters. Let me know what you think, or if I made any mistake.